File: java/lib/src/test/java/org/sammancoaching/library/LoanPeriodTest.java
package org.sammancoaching.library;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.function.Executable;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class LoanPeriodTest {

    @Test
    void nopeAgain() {
        int days = 0;

        Executable createLoanPeriod = () -> LoanPeriod.ofDays(days);

        assertThrows(IllegalArgumentException.class, createLoanPeriod);
    }

    @Test
    void nopeAgain2() {
        int days = -1;

        Executable createLoanPeriod = () -> LoanPeriod.ofDays(days);

        assertThrows(IllegalArgumentException.class, createLoanPeriod);
    }

    @Test
    void testDefaultLoanPeriod() {
        assertEquals(14, LoanPeriod.defaultPeriod().days());
    }

    @Test
    void testMaximumLoanPeriod() {
        assertEquals(30, LoanPeriod.ofDays(30).days());
    }

    @Test
    void tooMuch() {
        int days = 31;

        Executable createLoanPeriod = () -> LoanPeriod.ofDays(days);

        assertThrows(IllegalArgumentException.class, createLoanPeriod);
    }
}

File: csharp/Library.Tests/LoanPeriodTest.cs
using NUnit.Framework;
using SammanCoaching.Library;

namespace Library.Tests;

public class LoanPeriodTest
{
    [Test]
    public void nopeAgain()
    {
        var days = 0;

        Action act = () => LoanPeriod.OfDays(days);

        Assert.Throws<ArgumentException>(act);
    }

    [Test]
    public void nopeAgain2()
    {
        var days = -1;

        Action act = () => LoanPeriod.OfDays(days);

        Assert.Throws<ArgumentException>(act);
    }

    [Test]
    public void testDefaultLoanPeriod()
    {
        var expectedDays = 14;

        var loanPeriod = LoanPeriod.DefaultPeriod();

        Assert.That(loanPeriod.Days, Is.EqualTo(expectedDays));
    }

    [Test]
    public void testMaximumLoanPeriod()
    {
        var days = 30;

        var loanPeriod = LoanPeriod.OfDays(days);

        Assert.That(loanPeriod.Days, Is.EqualTo(days));
    }

    [Test]
    public void tooMuch()
    {
        var days = 31;

        Action act = () => LoanPeriod.OfDays(days);

        Assert.Throws<ArgumentException>(act);
    }
}
